app/src/main/java/com/edu/uni/augsburg/uniatron/stepcounter/StepCountService.java

```diff
@@ -0,0 +1,100 @@
+package com.edu.uni.augsburg.uniatron.stepcounter;
+
+import android.app.Service;
+import android.content.Intent;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.AsyncTask;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+import android.widget.Toast;
+
+//import com.edu.uni.augsburg.uniatron.MainApplication;
+
+/**
+ * The step count service collects steps and commits them to the database.
+ *
+ * @author Leon Wöhrl
+ */
+public class StepCountService extends Service implements SensorEventListener {
+
+    private SensorManager sensorManager;
+    private Sensor stepDetectorSensor;
+
+
+    private static int commitSize = 10;
+    private int currentSteps;
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        currentSteps = 0;
+    }
+
+    @Override
+    public int onStartCommand(Intent intent,   int flags, int startId) {
+        // grab step detector and register the listener
+        sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+        stepDetectorSensor = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_DETECTOR);
+        sensorManager.registerListener(this, stepDetectorSensor, SensorManager.SENSOR_DELAY_FASTEST);
+
+        Toast.makeText(this, "StepCountService has been started", Toast.LENGTH_SHORT).show();
```

Comment:
Remove every UI depending content. The service should be not visible to the user.